core/src/SYCL/Kokkos_SYCL.cpp

```diff
@@ -81,6 +81,13 @@ SYCL::SYCL() : m_space_instance(&Impl::SYCLInternal::singleton()) {
       "SYCL instance constructor");
 }
 
+SYCL::SYCL(sycl::queue const stream)
+    : m_space_instance(new Impl::SYCLInternal) {
```

Comment:
Who is in charge of calling `delete`?

Reply:
Good catch! Thanks! This should be fixed now. We employ the same strategy as for `Cuda` and `HIP` and now.
Unfortunately, this means that the `Policies` are no longer trivially copyable and I had toi make sure that we only copy (and use) relevant information to kernels.